extension/scripts/features/collapsibleareas/ttCollapsibleAreas.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+(async () => {
+	featureManager.registerFeature(
+		"Collapse Areas",
+		"sidebar",
+		() => settings.pages.sidebar.collapseAreas,
+		null,
+		addCollapseIcon,
+		removeCollapseIcon,
+		{
+			storage: ["settings.pages.sidebar.collapseAreas"],
+		},
+		null
+	);
+
+	async function addCollapseIcon() {
+		await requireSidebar();
+		if (!document.find("[class='header-arrow']") && !mobile) {
```

Comment:
Mobile check is in the `requirement` part of `registerFeature`.
[Example](https://github.com/Mephiles/torntools_2/blob/master/extension/scripts/features/computerlink/ttComputerLink.js)

Not sure what the element find is for, as it isn't used after the check, but it should probably go into the requirements as well.